outputs.tf

```diff
@@ -155,6 +155,11 @@ output "public_network_acl_arn" {
   value       = try(aws_network_acl.public[0].arn, null)
 }
 
+output "public_subnet_az_mapping" {
+  description = "A mapping of public subnet IDs to Availability Zones, useful when deploying PrivateLink endpoint services, which require matching AZs on both ends."
+  value       = { for subnet in aws_subnet.public[*] : (subnet["id"]) => subnet["availability_zone"] }
```

Comment:
rather than this custom mapping, I think a better approach would be to just output all attributes of the subnets created and let users use those as they please (you can use this to create this mapping for example in your own implementation)

```hcl
output "public_subnets" {
  description = "List of the subnets created and their associate attributes"
  value       = aws_subnet.public
}
```

Reply:
I'd built it this way to avoid a breaking change, hoping for a fast release, but I can arrange this too. Right now my org is working from my fork anyways. Alternatively, I could submit a second PR implementing the "breaking change" for merge in the next major version, your call!